thesis/03-implementation.tex

```diff
@@ -1,44 +1,43 @@
-From the list of features mentioned in the previous chapter we have selected the most important ones and created a prototype in a form of MVP (Minimum Viable Product). 
+From the list of requirements mentioned in the previous chapter we have selected the most important ones and created a prototype that would serve as a MVP (Minimum Viable Product) which we could develop further into a full-featured application.
+\smallskip
 
-The selected features:
+The selected features list consists of: 
 
 \begin{enumerate}
-	\requirement{3}{load an Instruction}
-	\requirement{3}{view the 3D representation of the Instruction step}
-	\requirement{3}{switch to the next step in the Instruction}
-	\requirement{3}{switch to the previous step in the Instruction}
-	\requirement{3}{see the Transition between two Instruction steps}
-	\requirement{2}{pause the Transition at any time}
-	\requirement{2}{rewind the Transition}
-	\requirement{2}{forward the Transition}
-	\requirement{3}{rotate the Model in 3D space}
-	\requirement{3}{zoom the Model in and out in 3D space}
-	\requirement{1}{see creases of the Model} 
-	\requirement{2}{distinguish paper's top and bottom sides} 
+	\requirement{3}{loading an Instruction}
+	\requirement{3}{viewing the 3D representation of the Instruction step}
+	\requirement{3}{switching to the next step in the Instruction}
+	\requirement{3}{switching to the previous step in the Instruction}
+	\requirement{3}{seeing the Transition between two Instruction steps}
+	\requirement{2}{pausing the Transition at any time}
+	\requirement{2}{rewinding the Transition}
+	\requirement{2}{forwarding the Transition}
+	\requirement{3}{rotating the Model in 3D space}
+	\requirement{3}{zooming the Model in and out in 3D space}
+	\requirement{1}{seeing creases of the Model} 
+	\requirement{2}{distinguishing paper's top and bottom sides} 
 \end{enumerate}
 
-Some of the listed functionalities require a numerical solver which is also included in the MVP.
-
-Both Instructions and Transitions are represented using \tech{.fold} files which we have extended with information required by the application.
-
 \subsubsection{Frontend}
 
 At first we had decided to use plain \tech{JavaScript} with the \tech{Rollup} bundler and \tech{Three.js} framework.
-It quickly became apparent that the lack of state management will become problematic as the development progresses. Following that realization we have decided to incorporate a reactive framework - \tech{React} to aid the project with basic layout and aforementioned state manipulation. 
-Due to minor interoperability issues the Rollup was also replaced with \tech{Webpack}.
+It quickly became apparent that the state management of the application will become problematic as the development progresses.
+Following that realization we have decided to incorporate a framework for building user interfaces - \tech{React} - to structure the project as a well-defined UI components and to help with the aforementioned state manipulation. 
```

Comment:
Maybe a comma at the end of "Following that realization". It seems to be an introductory sentence.